src/cpp/fastdds/domain/DomainParticipantImpl.cpp

```diff
@@ -625,6 +625,20 @@ const PublisherQos& DomainParticipantImpl::get_default_publisher_qos() const
     return default_pub_qos_;
 }
 
+const PublisherQos DomainParticipantImpl::get_publisher_qos_from_profile(
+        const std::string& profile_name) const
+{
```

Comment:
Change the method signature as explained above (apply also to `get_subscriber_qos_from_profile` and `get_topic_qos_from_profile`.
Follow the same logic used in `DomainParticipantFactory::get_participant_qos_from_profile`